stage5.2_fuller_application/pycasa/ui/image_file_view.py

```diff
@@ -0,0 +1,56 @@
+# General imports
+from matplotlib.figure import Figure
+from matplotlib import patches
+
+# ETS imports
+from traits.api import Button, Instance, observe
+from traitsui.api import HGroup, Item, ModelView, Spring, View
+
+# Local imports
+from ets_tutorial.util.mpl_figure_editor import MplFigureEditor
+from ..model.image_file import ImageFile
+
+
+class ImageFileView(ModelView):
+    """ ModelView for an image file object.
+    """
+    model = Instance(ImageFile)
+
+    figure = Instance(Figure)
+
+    detect_button = Button("Detect faces")
+
+    view = View(
+        Item("model.filepath", style="readonly", show_label=False),
+        Item("figure", editor=MplFigureEditor(), show_label=False),
+        HGroup(
+            Spring(),
+            Item("detect_button", show_label=False),
+            Spring(),
+        ),
+    )
+
+    @observe("model.filepath")
+    def build_mpl_figure(self, event):
+        figure = Figure()
+        axes = figure.add_subplot(111)
+        axes.imshow(self.model.to_array())
+        self.figure = figure
+
+    def _detect_button_fired(self):
+        self.model.detect_faces()
+
+    @observe("model.faces")
+    def update_mpl_figure_with_faces(self, events):
+        axes = self.figure.get_axes()[0]
+        for face in self.model.faces:
+            axes.add_patch(
+                patches.Rectangle(
+                    (face['c'], face['r']),
+                    face['width'],
+                    face['height'],
+                    fill=False,
+                    color='r',
+                    linewidth=2
+                )
+            )
```

Comment:
This is never going to remove rectangles once drawn.  If `model.faces` changes to a different set of faces, then this will redraw over any old ones, and add any new ones, but the patches for any that are no longer present will still be there.

This should either:
- remove all patches other than the base image and then add the rectangles for the face; or
- generate an entirely new figure

Reply:
Ah, thanks, great point!

Reply:
It felt wasteful, but maybe for simplicity's sake, I will just make a new figure for this PR...